### PR TITLE
feat: store Azure instance's IP address

### PR DESCRIPTION
**Pull request by ezr-ondrej**
Refactor Azure creation.
Extracts the common networking to be done once for all instances. Extracts Networking preparation to separate method, so we can store info about these.

Adjust Instance storing information to keep the IP address.

Refs HMS-1595

### CODE REVIEW

File: internal/clients/http/azure/create_vm.go [hunk 1]
```diff
@@ -85,7 +87,9 @@ func (c *client) WaitForVM(ctx context.Context, resumeToken string) (clients.Azu
 		span.SetStatus(codes.Error, "polling of virtual machine creation status failed to start")
 		return "", fmt.Errorf("polling of virtual machine creation status failed to start: %w", err)
 	}
-	resp, err := pollerResponse.PollUntilDone(ctx, nil)
+	resp, err := pollerResponse.PollUntilDone(ctx, &runtime.PollUntilDoneOptions{
+		Frequency: vmPollFrequency,
```
Comment (ezr-ondrej) [inline internal/clients/http/azure/create_vm.go:91]: slipped off in previous refactoring back to default :/

File: scripts/rest_examples/reservation-create-azure.http [hunk 2]
```diff
@@ -5,7 +5,7 @@ X-Rh-Identity: {{identity}}
 
 {
   "name": "azure-linux-us-east",
-  "source_id": "5",
+  "source_id": "{{azure-source-id}}",
```
Comment (lzap) [inline scripts/rest_examples/reservation-create-azure.http:8]: This needs to be added in the `http-client-env.json` file I think GoLand will error out once this is used. But it’s fine.

**Review by lzap — APPROVED** [review]
Sure.

**Comment by lzap** [comment]
@akhil-jha the API change is expected, instead of string we return struct with additional fields. This change was done previously for AWS.

**Review by akhil-jha — APPROVED** [review]

**Comment by ezr-ondrej** [comment]
Nice! Thanks for quick reviews :orange_heart:

File: scripts/rest_examples/reservation-create-azure.http (hunk 2 above)
Comment (ezr-ondrej) [reply scripts/rest_examples/reservation-create-azure.http:8]: I've got it defined locally, there is no good default value for the comitted context IMO :thinking:

**Pull request merged**